### ISSUE TITLE
aws-elasticloadbalancingv2-targets: InstanceTarget should register its instance as a connectable when added to a target group

### ISSUE THREAD

**Issue by luxaritas** [aws/aws-cdk#20753]
### Describe the feature

When adding an ASG target, it calls registerConnectable on the targetGroup so that the ASG will accept connections from the load balancer. it seems like instance targets should do the same

### Use Case

If I make an instance a target of a load balancer, almost certainly I want to allow connections from the load balancer to said instance. :)

### Proposed Solution

See the difference between https://github.com/aws/aws-cdk/blob/81ef6650d123726ee01ec6cecba77d37244290e4/packages/%40aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts#L1382-L1396 and https://github.com/aws/aws-cdk/blob/81ef6650d123726ee01ec6cecba77d37244290e4/packages/%40aws-cdk/aws-elasticloadbalancingv2-targets/lib/instance-target.ts#L26-L28

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.27.0

### Environment details (OS name and version, etc.)

Manjaro 21.3.0

**Comment by corymhall** [comment]
@luxaritas I agree, the instance behavior should be the same as the autoscaling group.

I am unassigning and marking this issue as p2, which means that we are unable to work on this immediately, but since this looks to be a simple change we would accept contributions.

**Comment by luxaritas** [comment]
At least, it think so? Would it actually make sense to introduce a "looked up instance" to handle this scenario? Or should this only be a feature on an InstanceTarget and not an InstanceIdTarget? Or should the allowed connection be added some other way?

Looking at how an instances `connections` property works, it seems like it specifies a default security group which wouldn't be able to be inferred from a lookup, so maybe it wouldn't make sense to support this on InstanceIdTarget?

**Comment by zweger** [comment]
I ran into this issue recently. CDK's current behavior is very surprising - why should I need to manually configure security group rules for my ALB only for InstanceTargets?
